src/main/objects/RulesChecker.js

```diff
@@ -181,7 +169,7 @@ function checkConnection(prevChord, currentChord){
                     break;
                 }
             }
-            if (dominantVoiceWith3 > -1 && currentChord.notes[dominantVoiceWith3].chordComponent.chordComponentString !== "1") return -1;
+            if (dominantVoiceWith3 > -1 && !prevChord.notes[dominantVoiceWith3].equals(currentChord.notes[dominantVoiceWith3]) && !Utils.containsChordComponent(currentChord.harmonicFunction.omit, "1") && currentChord.notes[dominantVoiceWith3].chordComponent.chordComponentString !== "1") return -1;
```

Comment:
To bardziej tak kosmetycznie, ale poprzenosiłbym to do nowych linii, bo się to nie mieści na monitorze ;)

Reply:
Done